libs/components/layout/testing/src/modules/back-to-top/back-to-top-harness.spec.ts

```diff
@@ -0,0 +1,30 @@
+import { TestbedHarnessEnvironment } from '@angular/cdk/testing/testbed';
+import { TestBed } from '@angular/core/testing';
+import { NoopAnimationsModule } from '@angular/platform-browser/animations';
+import { SkyAppTestUtility } from '@skyux-sdk/testing';
+
+import { SkyBackToTopHarness } from './back-to-top-harness';
+import { BackToTopHarnessTestComponent } from './fixtures/back-to-top-harness-test.component';
+
+describe('BackToTop test harness', () => {
+  it('should get the back to top component and click it', async () => {
+    await TestBed.configureTestingModule({
+      imports: [BackToTopHarnessTestComponent, NoopAnimationsModule],
+    }).compileComponents();
+
+    const fixture = TestBed.createComponent(BackToTopHarnessTestComponent);
+    const loader = TestbedHarnessEnvironment.documentRootLoader(fixture);
+
+    await expectAsync(loader.getHarness(SkyBackToTopHarness)).toBeRejected();
+
+    window.scrollTo(0, document.body.scrollHeight);
+    SkyAppTestUtility.fireDomEvent(window, 'scroll');
+    fixture.detectChanges();
+    await fixture.whenStable();
+
+    const backToTopHarness: SkyBackToTopHarness =
+      await loader.getHarness(SkyBackToTopHarness);
+
+    await backToTopHarness.clickBackToTop();
```

Comment:
we dont confirm anything about the click actually working. so from what i can tell, clicking the back to top, makes the button dissapear, meaning trying to access the harness would fail, can we check for that?

Reply:
from conversation: we dont have a great way of getting this information but i think following the unit test that adds an id to the topmost element on the page and checks if its bounding rectangle is inside the window height makes the most sense to me.

unsure if this is what we should show consumers tho.

Reply:
added this validation